### DISCUSSION TITLE
Proposed requests overhaul

### DISCUSSION THREAD

**Discussion by UberWaffe** [Keriew/augustus#884] [Ideas]
The idea is to overhaul requests to be a bit more like events.

A request consists of:
One or more conditions before the request is sent to the player. (By default there is one condition, which is time passed.)
A custom message to show to the player. (Default None, which then simply uses the current built-in city message.)
One or more requested goods. (Type + amount. Default is only 1 requested good type.)
Time in months allowed for fulfilling the request. (Default is in years.)
Time in months before first deadline that reminder is sent. (Default is 12 months, or half the allowed months if 1 year or less.)
Time in months that extension is given for. (Default is 12 months.)
One or more actions to take when the request is successfully completed. (By default, 1 action, which is adding favour.)
One or more actions to take when Caesar has to ask again and give extension. (By default, 1 action, which is reducing favour.)
One or more actions to take when request is completely ignored. (By default, 1 action, which is reducing favour.)

The conditions mentioned above are the same conditions that are available for use in events.
The actions mentioned above are the same actions that are available for use in events.

Requests should allow exporting to XML, and importing from XML the same as custom events do.

As a player, when playing a scenario and viewing a request via the UI, the UI should show me the results of fulfilling and failing the request.

I.e. For all of the action, show each in human readable terms. This same logic would also be usable for event chains.
Example: ACTION_TYPE_ADJUST_FAVOR with amount -5 on the failed request actions could display to the user as: "Caesar's favour changes by -5".

**Comment by crudelios** [comment]
I like this.

Also, one thing that could be added to both events and requests is the fulfilling of a main mission objective. 

That means if you fail the event, you lose the mission.
